ykrt/src/mt.rs

```diff
@@ -94,17 +99,6 @@ impl<I> Location<I> {
     }
 }
 
-impl<I> Drop for Location<I> {
-    fn drop(&mut self) {
-        let lp = self.state.load(Ordering::Relaxed);
-        if lp & PHASE_TAG == PHASE_COMPILING {
-            unsafe {
-                Box::from_raw((lp & !PHASE_TAG) as *mut CompilingTrace<I>);
```

Comment:
This is running on the only thread having access to the `Location` so there can be no race-condition. You could even use `self.state.get_mut()` instead of an atomic load.

Reply:
This hunk has been deleted, but, yes, if it's brought back, I agree.

Reply:
The removal causes a memory leak, right? Why was it removed? There couldn't be a race-condition.

Reply:
It does indeed (there's a comment on it earlier in the PR). However, there are several ways we leak at the moment, and I need a more principled way to handle them in one go.